modules/data.remote/R/remote_process.R

```diff
@@ -0,0 +1,1054 @@
+##' call rp_control (from RpTools Python package) and store the output in BETY
+##'
+##' @name remote_process
+##' @title remote_process
+##' @export
+##'
+##' @param settings PEcAn settings list containing remotedata tags: raw_mimetype, raw_formatname, source, collection, scale, projection, qc, algorithm, credfile, pro_mimetype, pro_formatname, out_get_data, out_process_data, overwrite
+##'
+##' @examples
+##' \dontrun{
+##' remote_process(settings)
+##' }
+##' @author Ayush Prasad, Istem Fer
+##'
+
+remote_process <- function(settings) {
+  # Information about the date variables used in remote_process:
+  # req_start, req_end : start, end dates requested by the user, the user does not have to be aware about the status of the requested file in the DB
+  # start, end : effective start, end dates created after checking the DB status. These dates are sent to rp_control for downloading and processing data
+  # write_raw_start, write_raw_end : start, end dates which are used while inserting and updating the DB
+  # the "pro" version of these variables have the same meaning and are used to refer to the processed file
+  
+  # The value of remotefile_check_flag denotes the following cases:
+  
+  # When processed file is requested,
```

Comment:
great, thanks for adding this, but is there a `case 0` when nothing needs to be done?

Reply:
Not really. In those cases, the `write_***_ start` and `write_***_end` contain `dont write` so nothing actually gets changed in the DB and the details of the existing data are returned. But maybe I could represent this using `0` case, I'll think about this.